IQSS/dataverse
# fix S3 direct upload NPE and NetCDF/HDF5 and keep geospatial metadata extraction

**What this PR does / why we need it**:

The fileMetadataExtractableFromNetcdf method added in PR #9541 breaks S3 direct upload. This PR fixes it.

It turns out a number of features don't work when S3 direct upload is enabled. This PR enumerates them.

Finally a config setting was added: dataverse.netcdf.geo-extract-s3-direct-upload.

**Which issue(s) this PR closes**:

- closes #9601
- related to #6762

**Special notes for your reviewer**:

As `if (driverType.equals("tmp"))` in IngestServiceBean.saveAndAddFilesToDataset() is executing, note that if S3 direct upload is enabled, driverType is "s3". However, if S3 direct upload is disabled (but S3 is still enabled), driverType is "tmp". (I found this confusing. I'd be happy to add a comment in the code about this.)

This means that methods like `extractMetadataNcml()` and code to generate thumbnails is not executed when S3 direct upload is enabled.

The behavior above is not changed by this pull request, but it did lead me to set expectations better in the guides.

**Suggestions on how to test this**:

Primary bug:

- enable S3 direct upload
- upload any file, should work 
- upload a NetCDF or HDF5 file, expect no NcML file (always) and no geospatial metadata extraction (unless you enable the new config setting: dataverse.netcdf.geo-extract-s3-direct-upload)

To test FITS extraction:

- enable S3 direct upload
- enable astro block
- upload a FITS file (expect no metadata extraction and the following error in the log: "fits is a filename/extension Dataverse doesn't know about. Consider adding it to the MimeTypeDetectionByFileExtension.properties file.")

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

No.

**Is there a release notes update needed for this change?**:

Yes, I updated the previous release note to mention the new setting.

**Additional documentation**:

I expanded our docs on what doesn't work during S3 direct upload, including NetCDF/HDF5 metadata extraction. I also documented the new setting.
